### DISCUSSION TITLE
Custom S3 IOManager without pickling?

### DISCUSSION THREAD

**Discussion by skeller88** [dagster-io/dagster#17711] [Q&A] [answered]
Hi Community,

I'm looking for an example of a custom S3 IOManager that can read/write unpickled csv and json files. It looks like I could write my own by copying https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws/s3/io_manager.py#L25 and making the file load step configurable by parameter, but I'm wondering if someone has done this already. 

Thanks,
Shane

**Comment by jamiedemaria** [comment, marked as answer]
Hey @skeller88  you can absolutely copy the file you linked and update the `load_from_path` and `dump_to_path` functions to not use pickle. I'm not aware of any examples that do this, but maybe some other community members will share things they've written!
